### PR TITLE
Updates Netlify Adapter docs for remote images

### PR DESCRIPTION
**Pull request by ascorbic**
#### Description (required)

https://github.com/withastro/adapters/pull/187 adds support to the Netlify adapter for `image.domains` and `image.remotePatterns`. This PR updates the docs with details of how to use them. It also fixes a couple of errors in the other code samples.

#### Related issues & labels (optional)

Depends on https://github.com/withastro/adapters/pull/187

- Closes #
- Suggested label:

### CODE REVIEW

File: src/content/docs/en/guides/integrations-guide/netlify.mdx [hunk 1]
```diff
@@ -157,7 +157,7 @@ To opt out of Netlify's Image CDN remote image optimization, use the `imageCDN`
 
 ```js title="astro.config.mjs" ins={8}
 import { defineConfig } from 'astro/config';
-import netlify from '@astrojs/netlify/functions';
+import netlify from '@astrojs/netlify';
```
Comment (alexanderniebuhr) [inline src/content/docs/en/guides/integrations-guide/netlify.mdx:160]: @ascorbic why the update of the import?

File: src/content/docs/en/guides/integrations-guide/netlify.mdx [hunk 2]
```diff
@@ -133,7 +133,7 @@ To implement redirects, access control or custom response headers for pre-render
...
```
Comment (alexanderniebuhr) [inline src/content/docs/en/guides/integrations-guide/netlify.mdx:136]: @ascorbic why the update of the import?

File: src/content/docs/en/guides/integrations-guide/netlify.mdx (hunk 1 above)
Comment (ascorbic) [reply src/content/docs/en/guides/integrations-guide/netlify.mdx:160]: The code sample was wrong. The import changed in a previous version.

File: src/content/docs/en/guides/integrations-guide/netlify.mdx [hunk 3]
```diff
@@ -168,13 +168,26 @@ export default defineConfig({
 });
 ```
 
-:::caution
-This adapter does not support the `image.domains` and `image.remotePatterns` config properties in your Astro config. To [specify remote paths for Netlify Image CDN](https://docs.netlify.com/image-cdn/overview/#remote-path), use the `remote_images` field in `netlify.toml`.
-:::
+If you are using images hosted on another domain, you must authorize the domain or URL patterns using the [`image.domains`](/en/reference/configuration-reference/#imagedomains) and [`image.remotePatterns`](/en/reference/configuration-reference/#imageremotepatterns) options:
```
Comment (sarah11918) [inline src/content/docs/en/guides/integrations-guide/netlify.mdx:171]:
```suggestion
If you are using images hosted on another domain, you must authorize the domain or URL patterns using the [`image.domains`](/en/reference/configuration-reference/#imagedomains) or [`image.remotePatterns`](/en/reference/configuration-reference/#imageremotepatterns) configuration options:
```
"and" sounds like they may have to configure both!

File: src/content/docs/en/guides/integrations-guide/netlify.mdx (hunk 1 above)
Comment (ascorbic) [reply src/content/docs/en/guides/integrations-guide/netlify.mdx:160]: See https://github.com/withastro/adapters/blob/main/packages/netlify/src/functions.ts

File: src/content/docs/en/guides/integrations-guide/netlify.mdx [hunk 4]
```diff
@@ -168,13 +168,26 @@ export default defineConfig({
...
+    output: 'server',
+    adapter: netlify(),
+    image: {
+      domains: ['example.com'],
+    },
+});
+```
+For more information, see [the guide to authorizing remote images](/en/guides/images/#authorizing-remote-images/). This is not required for images hosted on the same domain as your site.
 
 ### Static sites & Redirects
```
Comment (sarah11918) [inline src/content/docs/en/guides/integrations-guide/netlify.mdx:188]:
```suggestion
### Static sites with the Netlify Adapter
```
Now that this section is about MORE than redirects, what do we think about generalizing the section?

File: src/content/docs/en/guides/integrations-guide/netlify.mdx [hunk 5]
```diff
@@ -168,13 +168,26 @@ export default defineConfig({
...
 
-For static sites you usually don't need an adapter. However, if you use `redirects` configuration in your Astro config, the Netlify adapter can be used to translate this to the proper `_redirects` format.
+For static sites you usually don't need an adapter unless you are using the image service. However, if you use `redirects` configuration in your Astro config, the Netlify adapter can be used to translate this to the proper `_redirects` format.
```
Comment (sarah11918) [inline src/content/docs/en/guides/integrations-guide/netlify.mdx:190]:
```suggestion
For static sites (`output: 'static'`) hosted on Netlify, you usually don't need an adapter. However, some deployment features are only available through an adapter.

Static sites will need to install this adapter to use and configure Netlify's [image service](#netlify-mage-cdn-support).

If you use `redirects` configuration in your Astro config, the Netlify adapter can be used to translate this to the proper `_redirects` format.
```
Make this accurate, but I think this better sets up this section for "static sites that need to use this adapter"

**Review by sarah11918 — COMMENTED** [review]
Alright! Sorry this took so long to get to, and thank you so much for this amazing addition to the Netlify adapter, @ascorbic !

I made some suggestions below for you and @alexanderniebuhr to consider, but I will be happy when Alex is! 

Welcome to Team Docs! :partying_face:

File: src/content/docs/en/guides/integrations-guide/netlify.mdx (hunk 5 above)
Comment (ascorbic) [reply src/content/docs/en/guides/integrations-guide/netlify.mdx:190]: Yeah, that's definitely better

**Comment by ascorbic** [comment]
Thanks, @sarah11918!

File: src/content/docs/en/guides/integrations-guide/netlify.mdx [hunk 6]
```diff
@@ -168,13 +168,30 @@ export default defineConfig({
...
+});
+```
+For more information, see [the guide to authorizing remote images](/en/guides/images/#authorizing-remote-images/). This is not required for images hosted on the same domain as your site.
+
+### Static sites with the Netlify Adapter
+
+For static sites (`output: 'static'`) hosted on Netlify, you usually don't need an adapter. However, some deployment features are only available through an adapter.
 
-### Static sites & Redirects
+Static sites will need to install this adapter to use and configure Netlify's [image service](#netlify-mage-cdn-support).
```
Comment (alexanderniebuhr) [inline src/content/docs/en/guides/integrations-guide/netlify.mdx:192]:
```suggestion
Static sites will need to install this adapter to use and configure Netlify's [image service](#netlify-image-cdn-support).
```

File: src/content/docs/en/guides/integrations-guide/netlify.mdx [hunk 7]
```diff
@@ -168,13 +168,30 @@ export default defineConfig({
...
+export default defineConfig({
+    // ...
+    output: 'server',
+    adapter: netlify(),
+    image: {
+      domains: ['example.com'],
+    },
+});
+```
+For more information, see [the guide to authorizing remote images](/en/guides/images/#authorizing-remote-images/). This is not required for images hosted on the same domain as your site.
```
Comment (alexanderniebuhr) [inline src/content/docs/en/guides/integrations-guide/netlify.mdx:186]:
```suggestion
For more information, see [the guide to authorizing remote images](/en/guides/images/#authorizing-remote-images). This is not required for images hosted on the same domain as your site.
```

**Pull request merged**